src/core/interpreter/map-interpreter.ts

```diff
@@ -531,10 +546,14 @@ export class MapInterpreter<TInput extends NonPrimitive | undefined>
 
       return castToVariables(result);
     } catch (e) {
-      throw new JessieError('Error in Jessie script', e, {
-        node,
-        ast: this.ast,
-      });
+      if (e instanceof Error) {
+        throw new JessieError('Error in Jessie script', e, {
+          node,
+          ast: this.ast,
+        });  
+      }
+
+      throw e;
```

Comment:
That’s the messed up type in errors again. And I forgot to return it to original state and just suppress TSC as in other cases because I plan to look at error handling as a whole.